### ISSUE TITLE
[HOLD for payment 2024-06-06] [$250] mWeb - Taxes - Buttons on the top not filing the space evenly

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#41048]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.66-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @shawnborton 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1714046529643169

## Action Performed:
1. Resize the browser to mobile view 
2. Go to workspace settings > Taxes
3. Observe the top 2 buttons

## Expected Result:
The two buttons at the top should fill the space evenly

## Actual Result:
The two buttons are too wide and cause overflow from the wrapping container.

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence
![Screen Shot 2024-04-25 at 9 13 28 PM](https://github.com/Expensify/App/assets/38435837/b48905e6-ea91-431b-9a1c-1524b77d14b1)
![CleanShot 2024-04-25 at 08 05 31@2x](https://github.com/Expensify/App/assets/38435837/af05461e-c5b1-4927-92f3-3675562f24c2)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~018c24cec1ff92c41b</li>
        <li>Upwork Job ID: 1783687288261435392</li>
        <li>Last Price Increase: 2024-05-03</li>
<li>Automatic offers: </li>
<ul>
<li>DylanDylann | Reviewer | 0</li>
<li>Krishna2323 | Contributor | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @jliexpensify</details>

**Comment by shawnborton** [comment]
While we're at it, I think we might consider adjusting the space between the buttons here - maybe tighten it up to be 8px gap instead of 12px? Thoughts @Expensify/design?

12px:
![CleanShot 2024-04-25 at 21 26 09@2x](https://github.com/Expensify/App/assets/2319350/c7895b2b-975f-4087-83bf-ebba7c025b58)

8px:
![CleanShot 2024-04-25 at 21 26 34@2x](https://github.com/Expensify/App/assets/2319350/af62b5f5-3efd-4f37-9988-307309c4ce23)


And we'd wanna do that everywhere across all workspace pages.

**Comment by dubielzyk-expensify** [comment]
I'm down for `8px`

**Comment by jliexpensify** [comment]
@shawnborton not sure of the complexity of this issue (it feels easy, but I could be wrong), but if you feel like it should be less than $250, feel free to adjust the price!

**Comment by kirtiraj-malkhede-au6** [comment]
Hi Expensify team, 

I want to contribute in this issue.  

Please add me in your Slack Channel.

Email- kirtiraj.malkhede@outlook.com 

I already sent the email to **contributors@expensify.com** but still not received any invitation.

Looking to learn, collaborate, and contribute.

**Comment by dannymcclain** [comment]
@shawnborton @dubielzyk-expensify I'm not sure about adjusting the spacing. When I measure the gap in the product, it looks like it's `14px`, not 12.

![CleanShot 2024-04-26 at 08 42 55@2x](https://github.com/Expensify/App/assets/7516624/7763f62b-d8d9-41ba-a992-b0bf5df165cd)

Here's a comparison of _actually_ 12px vs. 8px

![image](https://github.com/Expensify/App/assets/7516624/c394cc2f-16c7-46c2-b1ea-7652081f3ac8)

Honestly both feel fine to me though. I just want to make sure we're making adjustments based on how things _should be_, not just based on how they look in the product (because that might be not quite right).

**Comment by dubielzyk-expensify** [comment]
> Honestly both feel fine to me though.

Yeah. They both look good. Personally probably prefer 8px visually, but given our footer button has 12px, maybe we go with that? I don't mind either way

**Comment by jliexpensify** [comment]
Not overdue, still discussing!

**Comment by shawnborton** [comment]
@dragnoir replying to your comments:

> when you select one of the options, the new button "1 selected", I think it should be stretched on all the screen width. What do you think?

Agree, it should be full width. I think we have a separate issue somewhere to update this button to be a dropdown button though, is that right @luacmartins?

> The header on those workspace pages, the space between top and buttons,, is defined by height value and it's equal to 80px

Yes, this is a separate inconsistency that we've talked about and will handle in a separate issue. One weird thing I noticed though is that there is padding on the left/right sides of the header, which is strange to me. 

> on the same screen, the table header "Status" is not aligned correctly

This is correct, the Status header is right-aligned. So the alignment is happening on the right side.

**Comment by shawnborton** [comment]
I think I still lean towards 8px gap as the standard... I'd love to make that a global rule if possible. Mostly thinking about what we plan to do with the Search page, and I think 8px feels a bit nicer there:
![CleanShot 2024-04-29 at 10 57 37@2x](https://github.com/Expensify/App/assets/2319350/7f5016ea-86e3-4e50-9a39-ae1e5a9a5969)

So I would love to just use 8px as the standard gap for horizontal button groupings. Thoughts? I don't feel strongly and could totally be convinced that 12px is the better gap.

**Comment by dannymcclain** [comment]
>I think I still lean towards 8px gap as the standard... I'd love to make that a global rule if possible. Mostly thinking about what we plan to do with the Search page, and I think 8px feels a bit nicer there
>So I would love to just use 8px as the standard gap for horizontal button groupings. Thoughts? I don't feel strongly and could totally be convinced that 12px is the better gap.

I am totally down for `8px` being the standard gap for horizontal button groupings. I just wanted to make sure we were sure 😂. Let's send it.

**Comment by luacmartins** [comment]
>Agree, it should be full width. I think we have a separate issue somewhere to update this button to be a dropdown button though, is that right @luacmartins?

Yes, this is the [issue](https://github.com/Expensify/App/issues/39832). We just merged the PR this morning though, so if we need to update any other styles, we should probably do it as part of this issue.

**Comment by shawnborton** [comment]
Nice, we tested and it looks like it works as expected on mobile.

**Comment by luacmartins** [comment]
Nice! So we can close this issue?

**Comment by DylanDylann** [comment]
@luacmartins @shawnborton Any update here?

**Comment by shawnborton** [comment]
I think this issue is different from the linked issue above, so I think we still need to solve this one. Just tested and it's still happening:
![CleanShot 2024-05-02 at 07 55 53@2x](https://github.com/Expensify/App/assets/2319350/944bd847-98c0-4f0a-bbcb-17a51f0c79d3)

**Comment by DylanDylann** [comment]
@shawnborton Let me sum up 
1. We need to update the space between 2 buttons to 8px (Also update in other workspaces page) as mentioned [here](https://github.com/Expensify/App/issues/41048#issuecomment-2082971063)
2. And apply some suggestions from @dragnoir as mentioned [here](https://github.com/Expensify/App/issues/41048#issuecomment-2082960964)

**Comment by shawnborton** [comment]
Correct, I think we only need to address your first bullet here.

**Comment by DylanDylann** [comment]
@Krishna2323's [proposal](https://github.com/Expensify/App/issues/41048#issuecomment-2078498139) looks good to me 
🎀👀🎀 C+ reviewed

**Comment by dubielzyk-expensify** [comment]
Do you have a link to the PR so I can test it out?

**Comment by dubielzyk-expensify** [comment]
It looks more correct now:
![CleanShot 2024-05-06 at 09 38 39@2x](https://github.com/Expensify/App/assets/145416827/c4af4ea3-f2b3-40f3-afd9-76beb9f7f3a0)

There is a 1px padding applied to these buttons which means that the spacing between the buttons becomes `10px` and not `8px`. Is this a hack for something I'm not aware of? 
![CleanShot 2024-05-06 at 09 39 36@2x](https://github.com/Expensify/App/assets/145416827/11f01af1-2cd9-44e4-a71a-60114c5e3210)

It also means that there's `21px` from the right edge to the button.

The same on desktop:
![CleanShot 2024-05-06 at 09 40 52@2x](https://github.com/Expensify/App/assets/145416827/875e8a77-9005-4ce9-a669-55559a8d51c5)

**Comment by shawnborton** [comment]
Whoa, that is a weird change. I feel like we should undo that 1px padding hack personally.

**Comment by dannymcclain** [comment]
Agree! Let's remove that 1px of weirdo padding if we can. That explains why [I was measuring 14px instead of 12px a bit ago](https://github.com/Expensify/App/issues/41048#issuecomment-2079435204).

**Comment by jliexpensify** [comment]
Payment Summry

- C: @Krishna2323 $250
- C+: @DylanDylann $250

[Upwork job](https://www.upwork.com/jobs/~018c24cec1ff92c41b)

Just waiting on the checklist.

**Comment by jliexpensify** [comment]
Bump @DylanDylann to complete the checklist please

**Comment by jliexpensify** [comment]
SIGH I have to create a new job because it's expired.

@DylanDylann and @Krishna2323 pleas accept: https://www.upwork.com/jobs/~013b50ebeebd734f43

**Comment by DylanDylann** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

 [@DylanDylann] The PR that introduced the bug has been identified. Link to the PR: **NA**
 [@DylanDylann] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **NA**
 [@DylanDylann] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **NA**
 [@DylanDylann] Determine if we should create a regression test for this bug. **Yes**
 [@DylanDylann] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.


**Regression Test Proposal**

1. Open workspace settings > Enable taxes, categories, distance rates and tags
2. Go taxes, categories, distance rates and tags page one by one
3. Verify the header buttons have 8px gap between them
- For small width devices
1. Open workspace settings > Enable taxes, categories, distance rates and tags
2. Go taxes, categories, distance rates and tags page one by one
3. Verify the header buttons have 8px gap between them and the buttons container has equal horizontal padding


Do we agree 👍 or 👎

**Comment by jliexpensify** [comment]
Paid and job closed, thanks everyone!
